### PR TITLE
jspecify.dev makeover

### PR DESCRIPTION
**Pull request by artempyanykh**
This is in continuation to #253. I've ported all content and a bit more.

Few things to note:
1. This version lacks Japanese translation. Docusaurus has [support for i18n](https://docusaurus.io/docs/i18n/introduction) but I'm not qualified to port the translation from Sphinx to Docusaurus.
2. I've added a blog with a 'Welcome' post. The blog post is there mainly to serve as a template for future publications. I can remove the post and a link to the blog if this is something we don't want to expose now.
3. I took liberty to put some 'Features' on the landing page. Obviously, the wording and the content of these features is something we want to adjust.

How to try it yourself:
* Option 1: checkout locally, cd to `docs`, run `npm start` (you need node!)
* Option 2: GH action produces an Artifact called documents (e.g. [here](https://github.com/jspecify/jspecify/actions/runs/2817956742)). Download it, unpack, run `python3 -m http.server 8888` or equivalent to browse the prebuilt website.

Screenshots:
Screenshot 2022-08-08 at 14 05 32
Screenshot 2022-08-08 at 13 18 47
Screenshot 2022-08-08 at 13 18 58
Screenshot 2022-08-08 at 13 19 06
Screenshot 2022-08-08 at 13 19 15

### CODE REVIEW

**Review by scolsen — COMMENTED** [review]
Some initial, small comments. This looks great! Thanks for working on this!

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Sphinx Document
+name: Docusaurus Docs
```
[the remark this answers is not in the document]
Comment (artempyanykh) [reply .github/workflows/docs.yml]: Fair! Let's leave it as just 'Docs'

File: docs/README.md [hunk 2]
```diff
@@ -1,52 +1,45 @@
...
 
-To build this document, run the following commands:
-
-```sh
-$ cd path/to/jspecify
-$ docker build -t jspecify-sphinx docs
-$ docker run --rm -v $(pwd)/docs:/docs jspecify-sphinx make linkcheck
-$ docker run --rm -v $(pwd)/docs:/docs jspecify-sphinx make html
+```
+$ npm install
```
[the remark this answers is not in the document]
Comment (artempyanykh) [reply docs/README.md:10]: Added this to the instruction, thanks!

File: docs/blog/2022-08-07-welcome.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+slug: welcome
+title: Welcome to JSpecify's Blog
```
[the remark this answers is not in the document]
Comment (artempyanykh) [reply docs/blog/2022-08-07-welcome.md]: Fixed, thanks!

**Comment by artempyanykh** [comment]
Thanks for taking a look @scolsen! Addressed and and pushed an update.

**Comment by kevinb9n** [comment]
The amount of time I've spent thinking about this may give the wrong impression of the appreciation I feel for it. :-) I give my blessing as long as I can ask you questions when I need to!

**Comment by artempyanykh** [comment]
Thanks @kevinb9n! Sure, happy to help :)

I did some more testing wrt GH action deployment, everything looked good. Current plan is to merge the branch on Monday; right now the rendered version is available in my fork if somebody wants to browse: https://artempyanykh.github.io/jspecify/

@KengoTODA I'm sorry that the new version lacks Japanese translation, but hopefully this is something we can improve going forward.

**Comment by raghsriniv** [comment]
> I did some more testing wrt GH action deployment, everything looked good. Current plan is to merge the branch on Monday; right now the rendered version is available in my fork if somebody wants to browse: https://artempyanykh.github.io/jspecify/

I love it, thanks for doing this.

**Comment by artempyanykh** [comment]
FYI the new website is live.

**Pull request merged**